components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java

```
@@ -0,0 +1,422 @@
+            } catch (IOException | JSONException e) {
+                throw new IdentityProvisioningException("Error while obtaining the access token from the response.", e);
+            }
+        } catch (IOException e) {
+            log.error("Error while closing HttpClient.");
+        }
+        return accessToken;
+    }
+
+    private JSONObject buildUserAsJson(ProvisioningEntity provisioningEntity) throws NullPointerException {
+
+        Map<String, String> requiredAttributes = getSingleValuedClaims(provisioningEntity.getAttributes());
+        String displayNameClaim = this.configHolder.getValue(Office365ConnectorConstants.OFFICE365_DISPLAY_NAME);
+        String mailNickNameClaim = this.configHolder.getValue(Office365ConnectorConstants.OFFICE365_EMAIL_NICKNAME);
+        String upnClaim = this.configHolder.getValue(Office365ConnectorConstants.OFFICE365_UPN);
+        String immutableIdClaim = this.configHolder.getValue(Office365ConnectorConstants.OFFICE365_IMMUTABLE_ID);
+        String ruleAttributeName = this.configHolder.getValue(Office365ConnectorConstants
+                .OFFICE365_MEMBERSHIP_ATTRIBUTE);
+        String ruleAttributeClaim = this.configHolder.getValue(Office365ConnectorConstants.OFFICE365_MEMBERSHIP_VALUE);
+        if (ruleAttributeClaim.isEmpty() && !ruleAttributeName.isEmpty()) {
+            ruleAttributeClaim = Office365ConnectorConstants.WSO2_ROLE_CLAIM;
+        }
+        String displayName = requiredAttributes.get(displayNameClaim);
+        String mailNickName = requiredAttributes.get(mailNickNameClaim);
+        String immutableId = requiredAttributes.get(immutableIdClaim);
+        String upn = requiredAttributes.get(upnClaim);
+        String ruleAttributeValue = requiredAttributes.get(ruleAttributeClaim);
+
+        if (displayName == null || mailNickName == null || immutableId == null || upn == null) {
+            throw new NullPointerException("One or more of the mandatory user attributes: display name, mail " +
```

Throw a specific exception instead of a `NullPointerException`